File: RentABikeWTR_v1_7.Model/Rezervacije.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RentABikeWTR_v1_7.Model
{
    public class Rezervacije
    {
        public int RezervacijaId { get; set; }
        public DateTime DatumIzdavanja { get; set; }
        public DateTime VrijemePreuzimanja { get; set; }
        public DateTime VrijemeVracanja { get; set; }
        public string? Napomena { get; set; }
        public double CijenaUsluge { get; set; }
        public bool StatusPlacanja { get; set; }
        public int? KorisnikID { get; set; }
        //public virtual Korisnici Korisnik { get; set; }

        public int BiciklID { get; set; }
        //public Bicikli Bicikl { get; set; }

        //public bool SaTuristicikimVodicem { get; set; }
        public int? TuristickiVodicID { get; set; }
        //public virtual TuristickiVodici TuristickiVodic { get; set; }


        public int? TuristRutaID { get; set; }
        //public virtual TuristRute TuristRuta { get; set; }


        //public int RezervacijaId { get; set; }

        public int KupacID { get; set; }
        //public Kupci Kupac { get; set; }
    }
}

File: rentabikewtr_mobile/lib/model/rezervacije.dart
import 'dart:ffi';
//import 'dart:html';

import 'package:json_annotation/json_annotation.dart';

part 'rezervacije.g.dart';

@JsonSerializable(explicitToJson: true)
class Rezervacije {
  int? rezervacijaId;
  DateTime? datumIzdavanja;
  DateTime? vrijemePreuzimanja;
  DateTime? vrijemeVracanja;
  bool statusPlacanja = true;
  @JsonKey(name: 'cijenaUsluge')
  num? cijenaUsluge;
  int? biciklId;
  int? turistickiVodicID;
  int? turistRutaID;
  int? kupacID;
  int? korisnikID;

  Rezervacije(
      {this.rezervacijaId,
      required datumIzdavanja,
      required vrijemePreuzimanja,
      required vrijemeVracanja,
      required this.statusPlacanja,
      this.cijenaUsluge,
      this.biciklId,
      this.turistickiVodicID,
      this.turistRutaID,
      this.kupacID,
      this.korisnikID}) {}

  factory Rezervacije.fromJson(Map<String, dynamic> json) =>
      _$RezervacijeFromJson(json);

  /// Connect the generated [_$RezervacijeToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RezervacijeToJson(this);
}
